From b2c5e7985c8bf5330fe1e61301434af5364c1503 Mon Sep 17 00:00:00 2001
From: Varun Vasudev <vvasudev@apache.org>
Date: Wed, 14 Oct 2015 02:55:21 -0700
Subject: [PATCH 1982/2848] YARN-4255. container-executor does not clean up
 docker operation command files. Contributed by
 Sidharta Seethana.

(cherry picked from commit da1ee078f9d3c2c25c51d0b392b0925821c42ad3)

(cherry picked from commit 2dd9c47)

Change-Id: I1cf13f6fbd0f43bb5e1668b43795900318c735bc
---
 .../container-executor/impl/container-executor.c   |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/container-executor.c b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/container-executor.c
index fe0a889..840d650 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/container-executor.c
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/native/container-executor/impl/container-executor.c
@@ -1081,6 +1081,9 @@ int run_docker(const char *command_file) {
   snprintf(docker_command_with_binary, EXECUTOR_PATH_MAX, "%s %s", docker_binary, docker_command);
   char **args = extract_values_delim(docker_command_with_binary, " ");
 
+  //clean up command file before we exec
+  unlink(command_file);
+
   int exit_code = -1;
   if (execvp(docker_binary, args) != 0) {
     fprintf(ERRORFILE, "Couldn't execute the container launch with args %s - %s",
@@ -1377,6 +1380,9 @@ int launch_docker_container_as_user(const char * user, const char *app_id,
   }
 
 cleanup:
+  //clean up docker command file
+  unlink(command_file);
+
   if (exit_code_file != NULL && write_exit_code_file(exit_code_file, exit_code) < 0) {
     fprintf (ERRORFILE,
       "Could not write exit code to file %s.\n", exit_code_file);
-- 
1.7.9.5

